class.leaflet-map.php

```diff
@@ -376,18 +376,32 @@ public function add_popup_to_shape($atts, $content, $shape)
         if (empty($message)) {
             return;
         }
-
+        
+        // save variable for filter
+        $original = $message;
+        
         // execute shortcodes if present:
         // e.g. [leaflet-marker][some-shortcode][/leaflet-marker]
         $message = do_shortcode($message);
+
+        // save variable for filter
+        $shortcoded = $message;
+
         $message = str_replace(array("\r\n", "\n", "\r"), '<br>', $message);
         $message = addslashes($message);
         $message = htmlspecialchars($message);
+        $message = "window.WPLeafletMapPlugin.unescape('{$message}')";
+
+        // use with: add_filter('leaflet_map_popup_message', 'example_callback');
+        // function takes default message, message after do_shortcode, and original/raw
+        $message = apply_filters('leaflet_map_popup_message', $message, $shortcoded, $original);
```

Comment:
@yonkov with this, hopefully you could do something like this:

```php
function example_callback( $message, $shortcoded, $original ) {
    return "`{$shortcoded}`";
}
add_filter( 'leaflet_map_popup_message', 'example_callback' );
```

so you'd have the shortcoded output, otherwise untouched.

Or you could even check if `$original` is, or has, a shortcode, before returning it like that.

Reply:
tested with the shortcode button I was testing and was able to output the raw shortcode:

<img width="379" alt="Screenshot 2023-02-19 at 10 13 58 PM" src="https://user-images.githubusercontent.com/1410985/219993730-4e9f4714-62ec-4f64-af80-ccd31958dc2e.png">

Reply:
And without the filter:

<img width="221" alt="Screenshot 2023-02-19 at 10 15 47 PM" src="https://user-images.githubusercontent.com/1410985/219994050-e82aafcb-fda5-40ed-91bb-546cefeed6ca.png">